src/routes/v2/slp.ts

```diff
@@ -2064,90 +2072,41 @@ const processInputs = (tx: TransactionInterface): any => {
-
-  // const vin = inputs.map(x => {
-  //   const slpAddress = x.e.a
-  //   const cashAddr: string = utils.toCashAddress(slpAddress)
-  //   const legacyAddr: string = utils.toLegacyAddress(slpAddress)
-
-  //   const hex: any = bitbox.Script.fromASM(x.str)
-
-  //   const obj: any = {}
-
-  //   obj.txid = x.e.h
-  //   obj.vout = x.e.i
-  //   obj.sequence = "n/a"
-  //   obj.n = x.i
-  //   obj.scriptSig = {
-  //     hex: hex.toString("hex"),
-  //     asm: x.str
-  //   }
-  //   obj.value = "n/a"
-  //   obj.legacyAddress = legacyAddr
-  //   obj.cashAddress = cashAddr
-  //   obj.slpAddress = slpAddress
-  //   return obj
-  // })
-
-  // const vout = outputs.map(x => {
-  //   const obj: any = {}
-  //   const hex: any = bitbox.Script.fromASM(x.str)
-
-  //   const val = new BigNumber(x.e.v / 100000000)
-
-  //   obj.value = val
-  //   obj.n = x.i
-  //   obj.scriptPubKey = {
-  //     hex: hex.toString("hex"),
-  //     asm: x.str
-  //   }
-  //   obj.spentTxId = "n/a"
-  //   obj.spentIndex = "n/a"
-  //   obj.spentHeight = "n/a"
-  //   return obj
-  // })
-
-  const sendOutputs: Array<string> = ["0"]
-  tokenOutputs.map(x => {
-    const string = parseFloat(x.amount) * 100000000
-    sendOutputs.push(string.toString())
-  })
+  try {
+    BigNumber.set({ DECIMAL_PLACES: 8 })
 
-  const obj = {
-    // txid: transaction.tx.h,
-    // version: 2,
-    // locktime: 0,
-    // vin: vin,
-    // vout: vout,
-    // blockhash: transaction.blk.h,
-    // blockheight: transaction.blk.i,
-    // confirmations: "n/a",
-    // time: transaction.blk.t,
-    // blocktime: transaction.blk.t,
-    // valueOut: "n/a",
-    // size: "n/a",
-    // valueIn: "n/a",
-    // fees: "n/a",
-    tokenInfo: {
-      versionType: transaction.slp.detail.versionType,
-      transactionType: transaction.slp.detail.transactionType,
-      tokenIdHex: transaction.slp.detail.tokenIdHex,
-      sendOutputs: sendOutputs
-    },
-    tokenIsValid: transaction.slp.valid
-  }
+    // Get the data from the unconfirmed or confirmed collection.
+    const transaction: any = slpDBFormat.data.u.length
+      ? slpDBFormat.data.u[0]
+      : slpDBFormat.data.c[0]
+
+    const inputs: Array<any> = transaction.in
 
-  return obj
+    const outputs: Array<any> = transaction.out
+    const tokenOutputs: Array<any> = transaction.slp.detail.outputs
+
+    const sendOutputs: Array<string> = ["0"]
+    tokenOutputs.map(x => {
+      const string = parseFloat(x.amount) * 100000000
+      sendOutputs.push(string.toString())
+    })
+
+    const obj = {
+      tokenInfo: {
+        versionType: transaction.slp.detail.versionType,
+        transactionType: transaction.slp.detail.transactionType,
+        tokenIdHex: transaction.slp.detail.tokenIdHex,
+        sendOutputs: sendOutputs
+      },
+      tokenIsValid: transaction.slp.valid
+    }
+
+    return obj
+  } catch (err) {
+    wlogger.error(`Error in slp.ts/formatToRestObject().`, err)
+
+    return false
```

Comment:
The error was generated when there is no SLP information in the confirmed or unconfirmed collections. e.g. when someone queries a non-SLP transaction. This PR adds the try/catch handler. Catch returns false, to indicate to the parent function that no SLP data was found.